### PR TITLE
5.4.0 changelog

### PR DESCRIPTION
**Pull request by slevenick**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+* containerattached: added `admin_groups` field to `google_container_attached_cluster` resource ([#6537](https://github.com/hashicorp/terraform-provider-google-beta/pull/6537))
+* dialogflowcx: added `advanced_settings` field to `google_dialogflow_cx_flow` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
+* dialogflowcx: added `advanced_settings` fields to `google_dialogflow_cx_page` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
+* dialogflowcx: added `advanced_settings`, `text_to_speech_settings`, `git_integration_settings` fields to `google_dialogflow_cx_agent` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
+* redis: allow `replica_count` to be set to zero in the `google_redis_cluster` resource ([#6534](https://github.com/hashicorp/terraform-provider-google-beta/pull/6534))
+* sql: added `ssl_mode` field to `google_sql_database_instance` resource ([#6564](https://github.com/hashicorp/terraform-provider-google-beta/pull/6564))
```
Comment (shuyama1) [inline CHANGELOG.md:28]:
```suggestion
```
since this change is reverted

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+* tpuv2: added more fields to `google_tpu_v2_vm` resource, including `network_config`, `scheduling_config`, `shielded_instance_config`, `service_account` and `data_disks` ([#6555](https://github.com/hashicorp/terraform-provider-google-beta/pull/6555))
+* tpuv2: added`accelerator_config` field to `google_tpu_v2_vm` resource ([#6559](https://github.com/hashicorp/terraform-provider-google-beta/pull/6559))
```
Comment (trodge) [inline CHANGELOG.md:30]:
```suggestion
* tpuv2: added `accelerator_config` field to `google_tpu_v2_vm` resource ([#6559](https://github.com/hashicorp/terraform-provider-google-beta/pull/6559))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,43 @@
 ## 5.4.0 (Unreleased)
 
+NOTES:
```
Comment (melinath) [inline CHANGELOG.md:3]:
```suggestion
DEPRECATIONS:
* bigquery: deprecated `cloud_spanner.use_serverless_analytics` on `google_bigquery_connection`. Use `cloud_spanner.use_data_boost` instead. ([#6539](https://github.com/hashicorp/terraform-provider-google-beta/pull/6539))

NOTES:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+
+FEATURES:
+* **New Data Source:** `google_cloud_identity_group_lookup` ([#6530](https://github.com/hashicorp/terraform-provider-google-beta/pull/6530))
+* **New Resource:** `google_network_connectivity_policy_based_route` ([#6552](https://github.com/hashicorp/terraform-provider-google-beta/pull/6552))
+* **New Resource:** `google_pubsub_schema_iam_*` ([#6533](https://github.com/hashicorp/terraform-provider-google-beta/pull/6533))
+
+IMPROVEMENTS:
+* accesscontextmanager: added support for Access Context Manager's VPC Sub-segmentation feature ([#6553](https://github.com/hashicorp/terraform-provider-google-beta/pull/6553))
+* apigee: added support for type in `google_apigee_environment` ([#6562](https://github.com/hashicorp/terraform-provider-google-beta/pull/6562))
+* bigquery: added data boost and parallelism input options to `google_bigquery_connection` ([#6539](https://github.com/hashicorp/terraform-provider-google-beta/pull/6539))
```
Comment (melinath) [inline CHANGELOG.md:18]:
```suggestion
* bigquery: added `cloud_spanner.database_role`, `cloud_spanner.use_data_boost`, and `cloud_spanner.max_parallelism` fields to `google_bigquery_connection` ([#6539](https://github.com/hashicorp/terraform-provider-google-beta/pull/6539))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,43 @@
...
+* container: added update support for `google_container_node_pool.node_config.taint` ([#6536](https://github.com/hashicorp/terraform-provider-google-beta/pull/6536))
+* containerattached: added `admin_groups` field to `google_container_attached_cluster` resource ([#6537](https://github.com/hashicorp/terraform-provider-google-beta/pull/6537))
+* dialogflowcx: added `advanced_settings` field to `google_dialogflow_cx_flow` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
+* dialogflowcx: added `advanced_settings` fields to `google_dialogflow_cx_page` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
+* dialogflowcx: added `advanced_settings`, `text_to_speech_settings`, `git_integration_settings` fields to `google_dialogflow_cx_agent` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
+* redis: allow `replica_count` to be set to zero in the `google_redis_cluster` resource ([#6534](https://github.com/hashicorp/terraform-provider-google-beta/pull/6534))
```
Comment (melinath) [inline CHANGELOG.md:27]: This should be a bug rather than an enhancement.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,43 @@
...
+* sql: added `ssl_mode` field to `google_sql_database_instance` resource ([#6564](https://github.com/hashicorp/terraform-provider-google-beta/pull/6564))
+* tpuv2: added more fields to `google_tpu_v2_vm` resource, including `network_config`, `scheduling_config`, `shielded_instance_config`, `service_account` and `data_disks` ([#6555](https://github.com/hashicorp/terraform-provider-google-beta/pull/6555))
```
Comment (melinath) [inline CHANGELOG.md:29]: could we maybe list all of the added fields instead of just a few of them?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,43 @@
...
+* tpuv2: added`accelerator_config` field to `google_tpu_v2_vm` resource ([#6559](https://github.com/hashicorp/terraform-provider-google-beta/pull/6559))
+
+BUG FIXES:
+* bigquery: fixed a bug when updating a `google_bigquery_dataset` that contained an `iamMember` access rule added out of band with Terraform ([#6550](https://github.com/hashicorp/terraform-provider-google-beta/pull/6550))
+* bigqueryreservation: fixed bug of incorrect resource recreation when `capacity_commitment_id` is unspecified in resource `google_bigquery_capacity_commitment` ([#6548](https://github.com/hashicorp/terraform-provider-google-beta/pull/6548))
+* cloudrunv2: set field `annotations` having all of annotations present on the resource in GCP in data source `google_cloud_run_v2_job` ([#6532](https://github.com/hashicorp/terraform-provider-google-beta/pull/6532))
```
Comment (melinath) [inline CHANGELOG.md:35]: Same change below.

```suggestion
* cloudrunv2: made `annotations` field on the `google_cloud_run_v2_job` data source include all annotations present on the resource in GCP ([#6532](https://github.com/hashicorp/terraform-provider-google-beta/pull/6532))
```

**Review by melinath — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,46 @@
...
+* cloudrunv2: marked `location` field as required in resource `google_cloud_run_v2_job`. Any configuration without `location` specified will cause resource creation fail ([#6540](https://github.com/hashicorp/terraform-provider-google-beta/pull/6540))
+* cloudrunv2: marked `location` field as required in resource `google_cloud_run_v2_service`. Any configuration without `location` specified will cause resource creation fail ([#6540](https://github.com/hashicorp/terraform-provider-google-beta/pull/6540))
+
+FEATURES:
+* **New Data Source:** `google_cloud_identity_group_lookup` ([#6530](https://github.com/hashicorp/terraform-provider-google-beta/pull/6530))
+* **New Resource:** `google_network_connectivity_policy_based_route` ([#6552](https://github.com/hashicorp/terraform-provider-google-beta/pull/6552))
+* **New Resource:** `google_pubsub_schema_iam_*` ([#6533](https://github.com/hashicorp/terraform-provider-google-beta/pull/6533))
+
+IMPROVEMENTS:
+* accesscontextmanager: added support for Access Context Manager's VPC Sub-segmentation feature ([#6553](https://github.com/hashicorp/terraform-provider-google-beta/pull/6553))
```
Comment (melinath) [inline CHANGELOG.md:19]: I think it might make sense to specify the fields that were added rather than the feature?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,46 @@
...
+* apigee: added support for type in `google_apigee_environment` ([#6562](https://github.com/hashicorp/terraform-provider-google-beta/pull/6562))
```
Comment (melinath) [inline CHANGELOG.md:20]:
```suggestion
* apigee: added support for `type` in `google_apigee_environment` ([#6562](https://github.com/hashicorp/terraform-provider-google-beta/pull/6562))
```

**Review by melinath — APPROVED** [review]

**Pull request merged**